### PR TITLE
feat: add riscv64/s390x/i386/i686 targets

### PR DESCRIPTION
**Pull request by YOU54F**
Add support for additional targets

- riscv64
- s390x
- i386
- i586
- i686
- ppc64le

All built fine on my machine

ProductName:            macOS
ProductVersion:         14.3.1
BuildVersion:           23D60

bar `mips64el` which requires xcode 15.1 (I'm currently running xcode 15.0)

(was trying to build cross musl targets for multiple rust platforms I am targeting)

Tested with cargo, for compiling rust apps from macos.

Requires rustup target adding, before compiling rust app 

`rustup target add x86_64-unknown-linux-musl`

Here is what my `.cargo/config` looks like

```
[target.x86_64-unknown-linux-musl]
linker = "x86_64-linux-musl-gcc"
[target.aarch64-unknown-linux-musl]
linker = "aarch64-linux-musl-gcc"
[target.arm-unknown-linux-musleabihf]
linker = "arm-linux-musleabihf-gcc"
[target.i686-unknown-linux-musl]
linker = "i686-linux-musl-gcc"
[target.i586-unknown-linux-musl]
linker = "i586-linux-musl-gcc"
[target.powerpc64le-unknown-linux-musl]
linker = "powerpc64le-linux-musl-gcc"
[target.s390x-unknown-linux-musl]
linker = "s390x-linux-musl-gcc"
[target.riscv64-unknown-linux-musl]
linker = "riscv64-linux-musl-gcc"
```

Built _rust_ apps appear to work on glibc based distros with no special flags required, at least with an initial smoke test of the binary

### CODE REVIEW

File: musl-cross.rb [hunk 1]
```diff
@@ -98,7 +103,9 @@ def install
     targets.push "aarch64-linux-musl" if build.with? "aarch64"
     targets.push "arm-linux-musleabihf" if build.with? "arm-hf"
     targets.push "arm-linux-musleabi" if build.with? "arm"
+    targets.push "i386-linux-musl" if build.with? "i386"
     targets.push "i486-linux-musl" if build.with? "i486"
```
Comment (FiloSottile) [inline musl-cross.rb:107]: Isn't i586 missing here?

**Comment by YOU54F** [comment]
Hey,

so it seems like for rust, the following are tier3 targets

- [target.powerpc64le-unknown-linux-musl]
- [target.s390x-unknown-linux-musl]
- [target.riscv64gc-unknown-linux-musl]

so there is no std library for them available to download with `rustup target add <target>`

Building for those targets with `-Z build-std` on `nightly` channel

`cargo +nightly build --release --target riscv64gc-unknown-linux-musl -Zbuild-std`

results in some errors

```
  = note: riscv64-linux-musl-gcc: error: crt1.o: No such file or directory
          riscv64-linux-musl-gcc: error: crti.o: No such file or directory
          riscv64-linux-musl-gcc: error: crtbegin.o: No such file or directory
          riscv64-linux-musl-gcc: error: crtend.o: No such file or directory
          riscv64-linux-musl-gcc: error: crtn.o: No such file or directory
```          

They actually reside in, so can hopefully just add in a linker path to include.

`/opt/homebrew/Cellar/musl-cross/0.9.9_2/libexec/riscv64-linux-musl/lib/`
`/opt/homebrew/Cellar/musl-cross/0.9.9_2/libexec/lib/gcc/riscv64-linux-musl/9.2.0`

File: musl-cross.rb (hunk 1 above)
Comment (YOU54F) [reply musl-cross.rb:107]: yep sorry, late omission.

**Comment by YOU54F** [comment]
So this is working nicely for me now

- I needed to build libunwind from source, was able to use these cross-compilers to do so
  - https://github.com/YOU54F/homebrew-libunwind-musl
- needed to update binutils to 2.38 for riscv64 in order to the `unsupported ISA subset `z'` error
  - https://github.com/YOU54F/homebrew-musl-cross-multilib/blob/ed5ebfda35e523460d25524a2a4b739a461f5c81/Formula/musl-cross-riscv64.rb#L92

I elected to provide separate formulas for each target, so they can be bottled separately - [link](https://github.com/YOU54F/homebrew-musl-cross-multilib/tree/main/Formula)

This is the way its done on another cross compilation repo 

https://github.com/messense/homebrew-macos-cross-toolchains

for me it makes sense, I'm likely to want to use these cross-compilers in CI systems, and I don't want to build from source for exotic targets.

Thanks for your work on the formula chap!

**Comment by YOU54F** [comment]
closing, will maintain via my own form

**Pull request closed without merging**